vpr/src/place/net_cost_handler.cpp

```diff
@@ -477,71 +563,27 @@ static void update_net_bb(const ClusterNetId& net,
     if (cluster_ctx.clb_nlist.net_sinks(net).size() < SMALL_NET) {
         //For small nets brute-force bounding box update is faster
 
-        if (bb_updated_before[net] == NetUpdateState::NOT_UPDATED_YET) { //Only once per-net
-            get_non_updatable_bb(net,
-                                 ts_bb_coord_new[net],
-                                 ts_layer_sink_pin_count[size_t(net)]);
+        if (pl_net_cost.bb_update_status[net] == NetUpdateState::NOT_UPDATED_YET) { //Only once per-net
+            bb_2d_3d_control_flow.get_non_updatable_bb(net);
         }
     } else {
         //For large nets, update bounding box incrementally
         int iblk_pin = tile_pin_index(blk_pin);
-        bool src_pin = cluster_ctx.clb_nlist.pin_type(blk_pin) == PinType::DRIVER;
 
         t_physical_tile_type_ptr blk_type = physical_tile_type(blk);
         int pin_width_offset = blk_type->pin_width_offset[iblk_pin];
         int pin_height_offset = blk_type->pin_height_offset[iblk_pin];
+        bool is_driver = bb_2d_3d_control_flow.is_driver(blk_type, blk_pin);
```

Comment:
Could leave the function if you like it, but I think only the if part is needed (no else).